core/conversion/var/Var.cpp

```diff
@@ -85,6 +87,32 @@ std::string Var::type_name() const {
   }
 }
 
+nvinfer1::ITensor* Var::ITensorOrFreeze(ConversionCtx* ctx) {
+  TRTORCH_CHECK(isITensor() || (isIValue() && ptr_.ivalue->isTensor()), "Requested either IValue containing a Tensor, or ITensor, however Var type is " << type_name());
```

Comment:
We might as well tell people the contained type too since the fail case is if it is an IValue but not containing a Tensor or the var is none. maybe something like
```c++
 ".. However Var type is " << type_name << (isIValue() ? "containing " << *(ptr_.ivalue->type()) : "");
```

Reply:
ptr_.ivalue->type() is a c10::TypePtr, it doesn't convert to a string easily without some proper checks. 

I think I would need to have something like:

``````

Reply:
I think that when you dereference the TypePtr it gives you a string we use this in the Var unwrapping functions too iirc

Reply:
https://github.com/NVIDIA/TRTorch/blob/fe06d0993d1ebf9083eada67a42e4fd46c4bee92/core/conversion/var/Var_inl.h#L12

Reply:
I think printing out the contents of the IValue (if it is an IValue) every time might be useful from a debugging standpoint. I've changed this accordingly